File: CS/WindowsFormsApplication5/UserControl1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace WindowsFormsApplication5
{
    public partial class UserControl1 : UserControl
    {
        public UserControl1()
        {
            InitializeComponent();
            SubscribeAllTheButtonsToOneButtonClickEvent();

        }

        private void SubscribeAllTheButtonsToOneButtonClickEvent()
        {
            foreach (Control control in layoutControl1.Controls)
            {
                if (control is SimpleButton)
                {
                    SimpleButton button = control as SimpleButton;
                    button.Click += button_Click;      
                }
            }
        }

        void button_Click(object sender, EventArgs e)
        {
            SimpleButton button = sender as SimpleButton;
            MessageBox.Show(button.Text + " button was clicked"); 
        }
    }
}

File: VB/WindowsFormsApplication5/UserControl1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace WindowsFormsApplication5
	Partial Public Class UserControl1
		Inherits UserControl
		Public Sub New()
			InitializeComponent()
			SubscribeAllTheButtonsToOneButtonClickEvent()

		End Sub

		Private Sub SubscribeAllTheButtonsToOneButtonClickEvent()
			For Each control As Control In layoutControl1.Controls
				If TypeOf control Is SimpleButton Then
					Dim button As SimpleButton = TryCast(control, SimpleButton)
					AddHandler button.Click, AddressOf button_Click
				End If
			Next control
		End Sub

		Private Sub button_Click(ByVal sender As Object, ByVal e As EventArgs)
			Dim button As SimpleButton = TryCast(sender, SimpleButton)
			MessageBox.Show(button.Text & " button was clicked")
		End Sub
	End Class
End Namespace
